src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,325 @@
+///  \brief         Chains up VectorNormsEstimator and NormalizeTransformerImpl
+///                 to normalize input vector data so that it's norm becoming 1
+///
+template <
+    typename InputT,
+    NormType NormT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class NormalizeEstimator :
+    public Components::PipelineExecutionEstimatorImpl<
+        Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+        Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::PipelineExecutionEstimatorImpl<
+            Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+            Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~NormalizeEstimator(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeEstimator);
+};
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  NormalizeTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(std::vector<std::double_t> norms) :
+    _norms(std::move(norms)),
+    _row(0) {
+        for (auto it = norms.begin(); it != norms.end(); ++it) {
+            if (*it < 0) {
+                throw std::invalid_argument("Norms shouldn't be less than 0!");
+            }
+        }
+}
+
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(Archive &ar) :
+    NormalizeTransformer(
+        [&ar](void) {
+            std::vector<std::double_t> norms(Traits<std::vector<std::double_t>>::deserialize(ar));
+
+            return NormalizeTransformer<InputT>(std::move(norms));
+        }()
+    ) {
+}
+
+template <typename InputT>
+void NormalizeTransformer<InputT>::save(Archive &ar) const /*override*/ {
+    Traits<std::vector<std::double_t>>::serialize(ar, _norms);

```

Comment:
>Traits<std::vector<std::double_t>>::serialize(ar, _norms); [](start = 4, length = 58)

Serialize the major and minor version (see other featurizers for an example of this).